src/NServiceBus.Transport.Msmq/InstanceMapping/InstanceMappingFileAccess.cs

```diff
@@ -8,10 +9,20 @@ class InstanceMappingFileAccess : IInstanceMappingFileAccess
     {
         public XDocument Load(string path)
         {
-            using (var file = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
-            using (var reader = XmlReader.Create(file))
+            if (Uri.TryCreate(path, UriKind.Absolute, out var uri) && !uri.IsFile)
```

Comment:
Is it worth getting rid of the `!` and reversing the order of the sections? i.e.

```csharp
if (Uri.TryCreate(path, UriKind.Absolute, out var uri) && uri.IsFile)
{
  // Manually create the FileStream
}
else
{
  // Just pass what we have to XmlReader.Create
}
```

If we cannot create a Uri currently we try and do FileStream path which seems more narrow than the XDocument Load.